src/pages/index.js

```diff
@@ -0,0 +1,291 @@
+imagePlus.src = plusSrc;
+
+const profileEditButton = document.querySelector(".profile__edit-btn");
+const editModal = document.querySelector("#edit-modal");
+
+const photoModal = document.querySelector("#photo-modal");
+const photoModalImage = photoModal.querySelector(".modal__photo");
+const photoModalCaption = photoModal.querySelector(".modal__description");
+
+const profileFormElement = document.forms["edit-profile"];
+const profileSaveBtn = profileFormElement.querySelector(".modal__save-button");
+
+const nameInput = profileFormElement.querySelector("#name");
+const jobInput = profileFormElement.querySelector("#description");
+
+const profileNameElement = document.querySelector(".profile__name");
+const profileJobElement = document.querySelector(".profile__description");
+
+api
+  .getAppInfo()
+  .then((result) => {
+    const savedCards = result[0];
+    const savedProfile = result[1];
+    imageAvatar.src = savedProfile["avatar"];
+    imageAvatar.alt = savedProfile["name"];
+    profileNameElement.textContent = savedProfile["name"];
+    profileJobElement.textContent = savedProfile["about"];
+    savedCards.forEach((card) => {
+      const cardEl = getCardElements(card);
+      cardsList.append(cardEl);
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", escapeListener);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", escapeListener);
+}
+
+function escapeListener(evt) {
+  const modal = document.querySelector(".modal_opened");
+  if (evt.key === "Escape") {
+    closeModal(modal);
+  }
+}
+
+const modalList = document.querySelectorAll(".modal");
+modalList.forEach((modal) => {
+  modal.addEventListener("click", function (evt) {
+    if (evt.target.parentElement.classList.contains("page")) {
+      closeModal(modal);
+    }
+  });
+});
+
+const closeButtons = document.querySelectorAll(".modal__close-button");
+closeButtons.forEach((button) => {
+  const popup = button.closest(".modal");
+  button.addEventListener("click", (evt) => {
+    if (button.classList.contains("modal__close-button_cancel")) {
+      evt.preventDefault();
+    }
+    closeModal(popup);
+  });
+});
+
+profileEditButton.addEventListener("click", () => {
+  openModal(editModal);
+
+  nameInput.value = profileNameElement.textContent;
+  jobInput.value = profileJobElement.textContent;
+
+  profileSaveBtn.disabled = true;
+  profileSaveBtn.classList.add("modal__save-button_disabled");
+});
+
+// The form submission handler. Note that its name
+// starts with a verb and concisely describes what it does.
+function handleProfileFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  const newName = nameInput.value;
+  const newJob = jobInput.value;
+  profileSaveBtn.textContent = "Saving...";
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions